### PR TITLE
Added `databricks_automatic_cluster_update_workspace_setting` resource

### PR DESCRIPTION
**Pull request by nkvuong**
## Changes
- Added `databricks_automatic_cluster_update_workspace_setting` resource

## Tests

- [x] `make test` run locally
- [x] relevant change in `docs/` folder
- [x] using Go SDK
- [x] tested manually

### CODE REVIEW

File: docs/resources/automatic_cluster_update_setting.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+* `enabled` - (Required) The configuration details.
+* `restart_even_if_no_updates_available` - (Optional) To force clusters and other compute resources to restart during the maintenance window regardless of the availability of a new update.
+
+A `maintenance_window` block that defines the maintenance frequency with the following arguments
+
+* A `week_day_based_schedule` block with the following arguments
+  * `day_of_week` - the day of the week in uppercase, e.g. `MONDAY` or `SUNDAY`
+  * `frequency` - one of the `FIRST_OF_MONTH`, `SECOND_OF_MONTH`, `THIRD_OF_MONTH`, `FOURTH_OF_MONTH`, `FIRST_AND_THIRD_OF_MONTH`, `SECOND_AND_FOURTH_OF_MONTH`, `EVERY_WEEK`.
+  * A `window_start_time` block that defines the time of your maintenance window
```
Comment (alexott) [inline docs/resources/automatic_cluster_update_setting.md:44]: Is it possible to specify a timezone? If no, what is the default timezone? API docs doesn't expose this information.

**Review by alexott — APPROVED** [review]

**Review by harshshah-db — APPROVED** [review]

**Comment by harshshah-db** [comment]
@nkvuong can we merge this if all looks good? I want to enable the public api docs.

File: settings/all_settings.go [hunk 2]
```diff
@@ -15,7 +15,8 @@ import (
 //  3. Add a new entry to the AllSettingsResources map below. The final resource name will be "databricks_<SETTING_NAME>_setting".
 func AllSettingsResources() map[string]common.Resource {
 	return map[string]common.Resource{
-		"default_namespace":         makeSettingResource[settings.DefaultNamespaceSetting, *databricks.WorkspaceClient](defaultNamespaceSetting),
-		"restrict_workspace_admins": makeSettingResource[settings.RestrictWorkspaceAdminsSetting, *databricks.WorkspaceClient](restrictWsAdminsSetting),
+		"default_namespace":                  makeSettingResource[settings.DefaultNamespaceSetting, *databricks.WorkspaceClient](defaultNamespaceSetting),
```
Comment (alexott) [inline settings/all_settings.go:18]: There is a conflict here that we need to solve.

File: settings/generic_setting.go [hunk 3]
```diff
@@ -301,6 +301,10 @@ func makeSettingResource[T, U any](defn genericSettingDefinition[T, U]) common.R
 			// are added for which setting_name is not computed, we'll need to expose this somehow as part of the setting
 			// definition.
 			s["setting_name"].Computed = true
+			// this customize func is needed for automatic_cluster_update_workspace setting
+			if _, ok := s["automatic_cluster_update_workspace"]; ok {
+				common.MustSchemaPath(s, "automatic_cluster_update_workspace", "enablement_details").Computed = true
+			}
```
Comment (alexott) [inline settings/generic_setting.go:307]: We need to provide ability to customize schema as part of generic functions, not put specific customization into the generic settings definitions

**Review by alexott — CHANGES_REQUESTED** [review]

File: settings/generic_setting.go [hunk 4]
```diff
@@ -141,6 +147,20 @@ func (w workspaceSetting[T]) GetId(t *T) string {
...
+	}
+	return func(s map[string]*schema.Schema) map[string]*schema.Schema {
+		s[etagAttrName].Computed = true
+		// Note: this may not always be computed, but it is for the default namespace setting. If other settings
+		// are added for which setting_name is not computed, we'll need to expose this somehow as part of the setting
+		// definition.
+		s["setting_name"].Computed = true
+		return s
+	}
+}
```
Comment (alexott) [inline settings/generic_setting.go:162]: Can we make it like this:

1. apply default schema customization by default
2. if custom function exist, apply it as well...

?

**Review by alexott — APPROVED** [review]

**Pull request merged**